upbound_uxp/README.md

```diff
@@ -0,0 +1,163 @@
+because of the guard rails that their platform teams
+implement with the Universal Crossplane.
+Its merit is the velocity increase and risk reduction
+while keeping options open when building cloud platforms.
+
+Businesses may offer their own internal
+cloud API to their teams through writing
+composite resource definitions. Their
+cloud can then map that API to downstream providers.
+The providers create and lifecycle manage the external
+cloud resources orchestrated through Universal
+Crossplane when a user claims resources.
+Amazon AWS, Microsoft Azure and Google
+GCP are out of the box supported cloud providers.
+
+This Datadog integration facilitates the monitoring
+of Universal Crossplane and provider Kubernetes pods.
+The value of observing the metrics is to be aware of
+the health of this self managed infrastructure provisioning
+and resource lifecycle management ecosystem to rest assured
+that the infrastructure is properly managed, or that actions
+may be needed to correct its configuration. The metrics
+inform about performance and resource consumption. This
+helps validating the Univeral Crossplane management cluster
+sizing and enables optimization.
+
+The checks monitor [Upbound UXP](https://docs.upbound.io/uxp/)
+related metrics.
+
+It looks for UXP and provider pods by default
+in the upbound-system Kubernetes
+namespace. The Upbound and provider
+pods emit metrics at port 8080/metrics in
+a Prometheus compatible format.
+
+## Setup
+
+### Prerequisites
+
+The Datadog agent requires permissions to discover UXP and provider pods.
+Create a service account, a cluster role, and a cluster role binding
+to grant those permissions.
+
+Note that the datadog agent in the configuration
+below resides in an arbitrary `monitoring` namespace.
+Your agent may reside in a namespace of your choice.
+Configure the service account and cluster role accordingly.
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: datadog-upbound
+  namespace: monitoring
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: apiserver-cluster-role
+  namespace: monitoring
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs: ["get", "list" ]
+  - nonResourceURLs: ["/metrics"]
+    verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: apiserver-cluster-role-binding
+subjects:
+- namespace: monitoring
+  kind: ServiceAccount
+  name: datadog-upbound
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: apiserver-cluster-role
+```
+
+### Installation
+
+To install the Upbound UXP check on your host:
+
+
+1. Install the [developer toolkit]
+(https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit)
+ on any machine.
+
+2. Run `ddev release build upbound_uxp` to build the package.
+
+3. [Download the Datadog Agent]["Datadog to supply the URL for the agent location"].
```

Comment:
```suggestion
```
I don't think this step is necessary to call out

Reply:
Removed.